client/config/testing.go

```diff
@@ -18,8 +18,18 @@ func TestClientConfig(t testing.T) (*Config, func()) {
 	conf.Node = mock.Node()
 	conf.Logger = testlog.HCLogger(t)
 
+	// On macOS, os.TempDir returns a symlinked path under /var which
+	// is outside of the directories shared into the VM used for Docker.
+	// Expand the symlink to get the real path in /private, which is ok.
+	dirName := os.TempDir()
+	tmpDir, err := filepath.EvalSymlinks(dirName)
+	if err != nil {
+		t.Fatalf("Could not create temporary directory %s: %v", tmpDir, err)
```

Comment:
Is this error message the right one? It seems like `TempDir` and `EvalSymlinks` don't actually create anything, and that we don't try that until `ioutil.TempDir`. Maybe "couldn't resolve temporary directory links"?